File: 692-top-k-frequent-words/692-top-k-frequent-words.py
class Solution:
    def topKFrequent(self, words: List[str], k: int) -> List[str]:
        d = {}
        
        for i in words:
            if i not in d.keys():
                d[i] = 1
            else:
                d[i] += 1
                
                
        arr = sorted(d.items(), key=lambda a: (-a[1], a[0]))
        
        ans = []
        for i in arr:
            if k==0: break
            ans.append(i[0])
            k-=1
            
        return ans

File: 692-top-k-frequent-words/692-top-k-frequent-words.cs
public class Solution {
    public IList<string> TopKFrequent(string[] words, int k) => words
        .GroupBy(w => w)
        .OrderByDescending(g => g.TryGetNonEnumeratedCount(out int c) ? c : g.Count())
        .ThenBy(g => g.Key)
        .Select(g => g.Key)
        .Take(k)
        .ToList();
}